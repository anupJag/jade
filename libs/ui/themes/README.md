# fe-themes

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `ng test fe-themes` to execute the unit tests via [Jest](https://jestjs.io).
