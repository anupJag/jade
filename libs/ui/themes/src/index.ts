export * from './lib/defaultTheme';
export * from './lib/styled';
export * from './lib/themeProvider';
export { default as styled } from './lib/styled';
export * from './lib/GlobalStyles';
