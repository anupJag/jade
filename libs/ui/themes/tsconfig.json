{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {
    "jsx": "react",
    "allowJs": false,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "types": ["node", "jest"]
  },
  "include": ["**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx"]
}
