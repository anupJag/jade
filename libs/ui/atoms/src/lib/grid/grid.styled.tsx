import { styled } from '@jade/ui/themes';
import { grid } from '@jade/ui/utils';
import { Box } from '@jade/ui/atoms';

export const Container = styled(Box)<{ isFluid?: boolean; withGutter?: boolean }>`
  ${props =>
    props.isFluid ? grid.containerFluid(props.withGutter) : grid.container(props.withGutter)}
`;

export const Row = styled(Box)<{ withGutter?: boolean }>`
  ${props => grid.row(props.withGutter)}
`;

export const Col = styled(Box)<{ columns?: grid.ColumnsCount; auto?: boolean }>`
  ${props => (props.columns ? grid.colWithCount(props.columns) : grid.col(props.auto))}
`;
