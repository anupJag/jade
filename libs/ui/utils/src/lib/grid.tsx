import { css } from '@emotion/core';
import { defaultTheme } from '@jade/ui/themes';

const { mediaQueries } = defaultTheme;

export type HalfColumnsCount = 1 | 2 | 3 | 4 | 5 | 6;
export type ColumnsCount = HalfColumnsCount | 7 | 8 | 9 | 10 | 11 | 12;
export type ColumnOrder = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
export type OffsetCount = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11;

const GRID_GUTTER = 30;
const GRID_GUTTER_COL_IMPACT = GRID_GUTTER / 2;
const GRID_GUTTER_COL_IMPACT_WITH_UNIT = `${GRID_GUTTER_COL_IMPACT}px`;
const GRID_GUTTER_COL_NEG_IMPACT_WITH_UNIT = `-${GRID_GUTTER_COL_IMPACT}px`;
const GRID_COUNT = 12;

const _calculateEvenColumnWidth = (cols: ColumnsCount) => parseFloat((100 / cols).toFixed(6));
const _calculateColumnWidth = (cols: ColumnsCount | OffsetCount) =>
  parseFloat(((cols * 100) / GRID_COUNT).toFixed(6));

export const containerFluid = (useGutter = true) => css`
  width: 100%;
  padding-right: ${useGutter ? GRID_GUTTER_COL_IMPACT_WITH_UNIT : 0};
  padding-left: ${useGutter ? GRID_GUTTER_COL_IMPACT_WITH_UNIT : 0};
  margin-right: auto;
  margin-left: auto;
`;

export const container = (useGutter = true) => css`
  ${containerFluid(useGutter)}
  ${mediaQueries.sm} {
    max-width: 488px;
  }
  ${mediaQueries.md} {
    max-width: 720px;
  }
  ${mediaQueries.lg} {
    max-width: 960px;
  }
  ${mediaQueries.xl} {
    max-width: 1044px;
  }
`;

export const row = (useGutter = true) => css`
  display: flex;
  flex-wrap: wrap;
  margin-right: ${useGutter ? GRID_GUTTER_COL_NEG_IMPACT_WITH_UNIT : 0};
  margin-left: ${useGutter ? GRID_GUTTER_COL_NEG_IMPACT_WITH_UNIT : 0};
`;

export const rowEvenCols = (colsCount: HalfColumnsCount, useGutter = true) => css`
  ${row(useGutter)}
  & > * {
    flex: 0 0 ${_calculateEvenColumnWidth(colsCount)}%;
    max-width: ${_calculateEvenColumnWidth(colsCount)}%;
  }
`;

export const col = (auto = true) => css`
  position: relative;
  width: 100%;
  padding-right: ${GRID_GUTTER_COL_IMPACT_WITH_UNIT};
  padding-left: ${GRID_GUTTER_COL_IMPACT_WITH_UNIT};
  ${auto
    ? `
  flex-basis: 0;
  flex-grow: 1;
  min-width: 0;
  max-width: 100%;
  `
    : null}
`;

export const colWithCount = (columns: ColumnsCount) => css`
  position: relative;
  width: 100%;
  padding-right: ${GRID_GUTTER_COL_IMPACT_WITH_UNIT};
  padding-left: ${GRID_GUTTER_COL_IMPACT_WITH_UNIT};
  flex: 0 0 ${_calculateColumnWidth(columns)}%;
  max-width: ${_calculateColumnWidth(columns)}%;
`;

export const colOrder = (columnOrder: ColumnOrder) => css`
  order: ${columnOrder};
`;

export const colOrderFirst = () => css`
  order: -1;
`;

export const colOrderLast = () => css`
  order: ${GRID_COUNT + 1};
`;

export const offset = (columns: OffsetCount) => css`
  margin-left: ${_calculateColumnWidth(columns)}%;
`;
