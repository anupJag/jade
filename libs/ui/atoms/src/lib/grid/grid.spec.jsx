import React from 'react';
import { render } from '@testing-library/react';

import { ThemeProvider } from '@jade/ui/themes';
import * as S from './grid';

describe('<Styled Component : Grid />', () => {
  let Grid;
  beforeEach(() => {
    Grid = render(
      <ThemeProvider>
        {/* With Default Values */}
        <S.Container>
          <S.Row>
            <S.Col></S.Col>
          </S.Row>
        </S.Container>
        {/* With Props :  snapshot check for props change in future */}
        <S.Container isFluid={true}>
          <S.Row withGutter={true}>
            <S.Col columns={1}></S.Col>
          </S.Row>
        </S.Container>
      </ThemeProvider>,
    );
  });

  /**
   * Capturing Sanpshot : Grid
   *
   *  To check style / structure changes in Styled Components
   */
  test("snapshot check for Grid's Styled components", () => {
    const { asFragment } = Grid;
    expect(asFragment()).toMatchSnapshot();
  });
});
