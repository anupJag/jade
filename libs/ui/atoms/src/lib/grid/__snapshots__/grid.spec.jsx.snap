// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Styled Component : Grid /> snapshot check for Grid's Styled components 1`] = `
<DocumentFragment>
  <div
    class="css-1q9iu3a"
  >
    <div
      class="css-1taa2i3"
    >
      <div
        class="css-1dmmwkm"
      />
    </div>
  </div>
  <div
    class="css-1wgdrzt"
  >
    <div
      class="css-1taa2i3"
    >
      <div
        class="css-1uoq67m"
      />
    </div>
  </div>
</DocumentFragment>
`;
